pkg/addon/configpolicy/agent_addon.go

```diff
@@ -115,7 +121,37 @@ func getValues(cluster *clusterv1.ManagedCluster,
 		}
 	}
 
-	if val, ok := addon.GetAnnotations()[prometheusEnabledAnnotation]; ok {
+	if val, ok := annotations[clientQPSAnnotation]; ok {
+		value, err := strconv.ParseUint(val, 10, 8)
+		if err != nil {
+			log.Error(err, fmt.Sprintf(
+				"Failed to verify '%s' annotation value '%s' for component %s (falling back to default value %d)",
+				clientQPSAnnotation, val, addonName, userValues.UserArgs.ClientQPS),
+			)
+		} else {
+			// This is safe because we specified the uint8 in ParseUint
+			userValues.UserArgs.ClientQPS = uint8(value)
+		}
+	} else { // not set explicitly
+		userValues.UserArgs.ClientQPS = userValues.UserArgs.EvaluationConcurrency * 15
+	}
+
+	if val, ok := annotations[clientBurstAnnotation]; ok {
+		value, err := strconv.ParseUint(val, 10, 8)
+		if err != nil {
+			log.Error(err, fmt.Sprintf(
+				"Failed to verify '%s' annotation value '%s' for component %s (falling back to default value %d)",
+				clientBurstAnnotation, val, addonName, userValues.UserArgs.ClientBurst),
+			)
+		} else {
+			// This is safe because we specified the uint8 in ParseUint
+			userValues.UserArgs.ClientBurst = uint8(value)
+		}
+	} else { // not set explicitly
+		userValues.UserArgs.ClientBurst = userValues.UserArgs.EvaluationConcurrency*22 + 1
```

Comment:
Was the `*22 + 1` recommended or did we make that up?

Reply:
It's the same as I put in config-policy-controller: https://github.com/open-cluster-management-io/config-policy-controller/blob/48f6ec26841a41ecd0cd3e24ef901a2bfe07ec47/main.go#L544

The controller-runtime default is 20 qps, 30 burst. @mprahl suggested `qps = concurrency * 15`, and this `*22 + 1` maintains the controller-runtime ratio, at least approximately.